cmd/clusterctl/cmd/topology.go

```diff
@@ -0,0 +1,306 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"sort"
+
+	"github.com/olekukonko/tablewriter"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster"
+	utilyaml "sigs.k8s.io/cluster-api/util/yaml"
+)
+
+type topologyDryRunOptions struct {
+	file    string
+	cluster string
+	outDir  string
+}
+
+var dr = &topologyDryRunOptions{}
+
+var topologyDryRunCmd = &cobra.Command{
+	Use:   "topology-dryrun",
+	Short: "Dry run changes to clusters that use managed topologies.",
+	Long: LongDesc(`
+	    Provide the list of objects that would be created, modified and deleted when a change is applied.
+		The objects that will be created and modified will be stored in a path passed using --outdir.
+	`),
+	Example: Examples(`
+	    # List the changes when modifying a cluster.
+		clusterctl alpha topology-dryrun -f modified-cluster.yaml --outdir output/
+
+		# List the changes of a single cluster when multiple clusters are affected.
+		clusterctl alpha topology-dryrun -f modified-cluster-class.yaml --cluster "cluster1" --outdir output/
+	`),
+	Args: cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return runTopologyDryRun()
+	},
+}
+
+func init() {
```

Comment:
Don't we need a "kubeconfig" and "namespace" flag? If I understood correctly the command currently just uses the default kubeconfig and the current namespace from there? (which would mean that the current namespace in the kubeconfig file must be set/changed to work with clusters which are not in default)

Reply:
Agree regarding namespace. I was wondering regarding kubeconfig, because a lot of other commands specify a `kubeconfig` flag